docs/book/src/developer/testing.md

```diff
@@ -48,6 +48,19 @@ Kubernetes objects and waiting for the controllers to take action. See the [quic
 
 Also in case of integration tests, considerations about [mocking external APIs](#mocking-external-apis) and usage of [generic providers](#generic-providers) apply. 
 
+
+## Fuzzing tests
+Fuzzing tests automatically inject randomly generated inputs, often invalid or with unexpected values, into functions to discover vulnerabilities. 
+
+Two different types of fuzzing are currently being used on the Cluster API repository:
+### Fuzz testing for API conversion
+Cluster API uses Kubernetes' conversion-gen to automate the generation of functions to convert our API objects between versions. These conversion functions are tested using the [FuzzTestFunc util in our conversion utils package](https://github.com/kubernetes-sigs/cluster-api/blob/1ec0cd6174f1b860dc466db587241ea7edea0b9f/util/conversion/conversion.go#L194).
+For more information about these conversions see the API conversion code walkthrough in our [video walkthrough series](./guide.md#videos-explaining-capi-architecture-and-code-walkthroughs).
+
+### OSS-Fuzz continuous fuzzing
```

Comment:
@AdamKorcz Let me know if you're happy with this section 🙂

Reply:
LGTM.